File: C/1780-check-if-number-is-a-sum-of-powers-of-three/main.go
package main

import "fmt"

func checkPowersOfThree(n int) bool {
	m := n
	steps := 0
	M := map[int]int{}
	for m > 0 {
		if m%3 != 0 {
			m--
			_, ok := M[steps]
			if ok {
				return false
			} else {
				M[steps]++
			}
		} else {
			m /= 3
			steps++
		}
	}
	return true
}

func main() {
	fmt.Println("")
}

File: C/1780-check-if-number-is-a-sum-of-powers-of-three/main.rs
struct Solution {}

impl Solution {
  pub fn check_powers_of_three(n: i32) -> bool {
    let mut m = n;
    while m > 0 {
      if m % 3 == 2 {
        return false;
      }
      m /= 3;
    }
    true
  }
}
